x-pack/plugins/secretstore/server/lib/crypt_keeper.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import crypto from 'crypto';
+import hash from 'object-hash';
+
+export function buildCrypt({ key }: { key: string | Buffer }) {
```

Comment:
Did you consider using https://github.com/elastic/node-crypto?

Reply:
Yes, I had but it does not support passing a aad, I was considering extending it to allow for using an aad.